### PR TITLE
feat(monitoring): add prometheusrules for alerts

### LINKED ISSUE #48: Improve Calico monitoring (operated and not operated)
For a general improvement on the monitoring side for the fury packages, we need to add alerts and dashboards on Calico, operated and not operated)

### PR DESCRIPTION
**Pull request by ralgozino**
This PR includes the following changes:

- add prometheusRules for alerts on errors in both Calico and Typha
- update maintenance guide with relevant information
- update readme with monitoring information
- add screenshots
- fix typo in Typha dashboard title and add relevant tags

Please notice that upstream (calico) does not provide official prometheus rules (alerts), only metrics. The ones being added are insipired on the little I could find online, tweaked and checked that made sense with the metrics that calico exposes.

Fixes #48

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,6 +87,32 @@ resources:
...
+<a href="docs/images/screenshots/calico-typha-dashboard.png"><img src="docs/images/screenshots/calico-typha-dashboard.png" width="250"/></a>
+<!-- markdownlint-enable MD033 -->
+
+> click on each screenshot for the full screen version
+
+The following set of alerts is included with the networking module:
+
+| Alert Name                     | Summary                                                                 | Description                                                                                                                       |
+| ------------------------------ | ----------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------- |
+| CalicoDatapaneFailuresHigh     | A high number of dataplane failures within Felix are happening          | Calico node pod {{ $labels.pod }} ({{ $labels.instance }}) has seen {{ $value }} dataplane failures within the last hour          |
```
Comment (nutellinoit) [inline README.md:107]:
```suggestion
| CalicoDataplaneFailuresHigh     | A high number of dataplane failures within Felix are happening          | Calico node pod {{ $labels.pod }} ({{ $labels.instance }}) has seen {{ $value }} dataplane failures within the last hour          |
```

File: katalog/calico/MAINTENANCE.md [hunk 2]
```diff
@@ -52,3 +52,24 @@ export CALICO_VERSION=3.24
...
+
+- <https://platform9.com/docs/kubernetes/catapult-rules-alarms#calico>
+- <https://platform9.com/docs/kubernetes/calico-monitoring>
+- <https://docs.sysdig.com/en/docs/sysdig-monitor/monitoring-integrations/application-integrations/calico/>
+- <https://docs.sysdig.com/en/docs/sysdig-monitor/monitoring-integrations/application-integrations/calico/#errors>
+
+You can generate a markdown table with the rules to include it in the readme with the following commad:
+
+```bash
+ yq e '.spec.groups[] | .rules[] |  "| " + .alert + " | " + (.annotations.summary // "-" | sub("\n",". "))+ " | " + (.annotations.description // "-" | sub("\n",". ")) + " |"' katalog/calico/monitoring/prometheusrules.yaml
```
Comment (nutellinoit) [inline katalog/calico/MAINTENANCE.md:74]: Nice!

File: katalog/calico/monitoring/prometheusrules.yaml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  name: calico-alerts
+  namespace: kube-system
+spec:
+  groups:
+    - name: calico.rules
+      rules:
+        # Calico Node (Felix)
+        # https://projectcalico.docs.tigera.io/reference/felix/prometheus
+
+        - alert: CalicoDatapaneFailuresHigh
```
Comment (nutellinoit) [inline katalog/calico/monitoring/prometheusrules.yaml:21]:
```suggestion
        - alert: CalicoDataplaneFailuresHigh
```

File: katalog/tigera/on-prem/monitoring/prometheusrules.yaml [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  name: calico-alerts
+  namespace: kube-system
+spec:
+  groups:
+    - name: calico.rules
+      rules:
+        # Calico Node (Felix)
+        # https://projectcalico.docs.tigera.io/reference/felix/prometheus
+
+        - alert: CalicoDatapaneFailuresHigh
```
Comment (nutellinoit) [inline katalog/tigera/on-prem/monitoring/prometheusrules.yaml:21]:
```suggestion
        - alert: CalicoDataplaneFailuresHigh
```

**Review by nutellinoit — CHANGES_REQUESTED** [review]
Minor changes, overall LGTM! Nice job 🚀

File: README.md (hunk 1 above)
Comment (ralgozino) [reply README.md:107]: nice catch! thanks, fixed.

**Review by nutellinoit — APPROVED** [review]
LGTM

**Pull request merged**